dotCMS/src/main/java/com/dotcms/storage/DataBaseStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,721 @@
+                    .setSQL("SELECT * FROM storage_group WHERE group_name = ?")
+                    .addParam(groupNameLC).loadObjectResults(this.getConnection())).getOrElse(
+                    Collections::emptyList);
+            result.setValue(!results.isEmpty());
+        });
+
+        return result.booleanValue();
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName  {@link String}
+     * @param objectPath {@link String}
+     * @return
+     */
+    @Override
+    public boolean existsObject(final String groupName, final String objectPath) {
+        final String groupNameLC = groupName.toLowerCase();
+        final String objectPathLC = objectPath.toLowerCase();
+        final MutableBoolean result = new MutableBoolean(false);
+
+        this.wrapCloseConnection(() -> {
+
+            final List results = Try.of(() -> new DotConnect()
+                    .setSQL("SELECT * FROM storage WHERE group_name = ? AND path = ?")
+                    .addParam(groupNameLC).addParam(objectPathLC)
+                    .loadObjectResults(this.getConnection())).getOrElse(Collections::emptyList);
+            result.setValue(!results.isEmpty());
+        });
+
+        return result.booleanValue();
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName {@link String} group name
+     * @return
+     */
+    @Override
+    public boolean createGroup(final String groupName) {
+        return createGroup(groupName, ImmutableMap.of());
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName    {@link String} group name
+     * @param extraOptions {@link Map} depending on the implementation it might need extra options or not.
+     * @return
+     */
+    @Override
+    public boolean createGroup(final String groupName, final Map<String, Object> extraOptions) {
+        final String groupNameLC = groupName.toLowerCase();
+        return this.wrapInTransaction(() ->
+                Try.of(() -> {
+                    new DotConnect().setSQL(" INSERT INTO storage_group (group_name) VALUES (?) ")
+                            .addParam(groupNameLC).loadResult(
+                            this.getConnection());
+                    return true;
+                }).getOrElseGet(throwable -> false)
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName {@link String} group name
+     * @return
+     */
+    @Override
+    public int deleteGroup(final String groupName) {
+        final String groupNameLC = groupName.toLowerCase();
+        return this.wrapInTransaction(() -> {
+            final Connection connection = this.getConnection();
+            final DotConnect dotConnect = new DotConnect();
+
+            final List<Map<String, Object>> maps = dotConnect
+                    .setSQL("SELECT hash FROM storage WHERE group_name = ?").addParam(groupNameLC)
+                    .loadObjectResults();
+
+            int count = deleteObjects(maps.stream().map(map -> map.get("hash").toString())
```

Comment:
Codacy found an issue: [The String literal "hash" appears 4 times in this file; the first occurrence is on line 321](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)